src/location.rs

```diff
@@ -0,0 +1,101 @@
+use serde::{Deserialize, Serialize};
+use std::{fs, io};
+use std::ffi::{OsString};
+use std::fs::{DirEntry};
+use std::path::{Path, PathBuf};
+use regex::Regex;
+use cached::cached_result;
+use crate::config::Location;
+
+use super::config;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct LocationEntry {
+    pub packet: String,
+    pub time: f64,
+    pub hash: String,
+}
+
+const ID_REG: &str = "^([0-9]{8}-[0-9]{6}-[[:xdigit:]]{8})$";
+cached_result! {
+    ENTRY_CACHE: cached::UnboundCache<PathBuf, LocationEntry> = cached::UnboundCache::new();
```

Comment:
Caching here is ok because these will never change. This is an indefinite, in-memory cache.

Reply:
Yeah I had the same questions. Probably some sized cache makes sense, boots out the oldest entries: https://docs.rs/cached/latest/cached/stores/struct.SizedCache.html But wasn't sure what size to go for?

Reply:
Let's leave as you have it for now and see how it goes